frontends/infinite-corridor/src/api/learning-resources/hooks.ts

```diff
@@ -20,15 +26,102 @@ const useUserList = (id: number) => {
 }
 
 const useUserListsListing = (options?: UserListOptions) => {
-  const url = urls.userListsListing(options)
-  const key = keys.userListsListing(options)
+  const url = urls.userList.listing(options)
+  const key = keys.userList.listing.page(options)
   return useQuery<PaginatedResult<UserList>>(key, () =>
     axios.get(url).then(res => res.data)
   )
 }
 
 const useUserListItems = (listId: number, options?: PaginationSearchParams) => {
-  return useQuery<PaginatedUserListItems>(urls.userListItems(listId, options))
+  const key = keys.userList.id(listId).itemsListing(options)
+  const url = urls.userList.itemsListing(listId, options)
+  return useQuery<PaginatedUserListItems>(key, () =>
+    axios.get(url).then(res => res.data)
+  )
+}
+
+const useFavorites = (options?: PaginationSearchParams) => {
+  const url = urls.favorite.listing(options)
+  const key = keys.favorites.listing.page(options)
+  return useQuery<PaginatedUserListItems>(key, () =>
+    axios.get(url).then(res => res.data)
+  )
+}
+
+const useTopics = () => {
+  const key = keys.topics
+  const url = urls.topics.listing
+  return useQuery<PaginatedResult<CourseTopic>>(key, () =>
+    axios.get(url).then(res => res.data)
+  )
 }
 
-export { useResource, useUserListItems, useUserList, useUserListsListing }
+const updateUserList = async (data: Partial<UserList> & { id: number }) => {
+  const url = urls.userList.details(data.id)
+  const { data: response } = await axios.patch(url, data)
+  return response
+}
+const useUpdateUserList = () => {
+  const queryClient = useQueryClient()
+  return useMutation(updateUserList, {
+    onSuccess: (_data, variables) => {
+      queryClient.invalidateQueries({
+        queryKey: keys.userList.id(variables.id).details
+      })
+      queryClient.invalidateQueries({
+        queryKey: keys.userList.listing.all
+      })
```

Comment:
The `onSuccess` logic is important and has no direct tests in this PR. It's responsible for invalidating the cached queries. I did add some direct testings for the onSuccess logic in [cc/populate-userlists](https://github.com/mitodl/open-discussions/blob/cc/populate-userlists/frontends/infinite-corridor/src/api/learning-resources/hooks.test.tsx).

We could also add an `onMutate` callback here for optimistic updates.